Test/CoreSDK.Test/Net46/Extensibility/Implementation/RichPayloadEventSourceTest.cs

```diff
@@ -95,61 +102,116 @@ public void RichPayloadEventSourcePageViewSentTest()
-                    string key = (string)tag["Key"];
-                    object value = tag["Value"];
-                    if (!string.IsNullOrWhiteSpace(key))
+                    Assert.IsNotNull(actualEvent);
+                    Assert.AreEqual(client.InstrumentationKey, actualEvent.Payload[0]);
+
+                    int keysFound = 0;
+                    object[] tags = actualEvent.Payload[1] as object[];
+                    foreach (object tagObject in tags)
                     {
-                        if (key == "ai.user.id")
-                        {
-                            Assert.AreEqual("testUserId", value);
-                            ++keysFound;
-                        }
-                        else if (key == "ai.operation.id")
+                        Dictionary<string, object> tag = (Dictionary<string, object>)tagObject;
+                        Assert.IsNotNull(tag);
+                        string key = (string)tag["Key"];
+                        object value = tag["Value"];
+                        if (!string.IsNullOrWhiteSpace(key))
                         {
-                            Assert.AreEqual(item.Context.Operation.Id, value);
-                            ++keysFound;
+                            if (key == "ai.user.id")
+                            {
+                                Assert.AreEqual("testUserId", value);
+                                ++keysFound;
+                            }
+                            else if (key == "ai.operation.id")
+                            {
+                                Assert.AreEqual(item.Context.Operation.Id, value);
+                                ++keysFound;
+                            }
                         }
                     }
+
+                    Assert.AreEqual(2, keysFound);
+                    Assert.IsNotNull(actualEvent.Payload[2]);
+
+                    var expectedPropertyNames = dataType.GetProperties().Select(i => i.Name);
+                    var actualPropertyNames = ((IDictionary<string, object>)actualEvent.Payload[2]).Keys.AsEnumerable();
+
+                    Assert.IsTrue(!expectedPropertyNames.Except(actualPropertyNames).Any() && !actualPropertyNames.Except(expectedPropertyNames).Any());
                 }
+            }
+            else
+            {
+                // 4.5 doesn't have RichPayload events
+                Assert.IsNull(RichPayloadEventSource.Log.EventSourceInternal);
+            }
+        }
 
-                Assert.AreEqual(2, keysFound);
-                Assert.IsNotNull(actualEvent.Payload[2]);
+        private static bool IsRunningOnVersion46AndAbove()
+        {
+            string productVersionString = System.Diagnostics.FileVersionInfo.GetVersionInfo(typeof(object).Assembly.Location).ProductVersion;
+
+            Version ver;
+            if (!Version.TryParse(productVersionString, out ver))
+            {
+                // Default is 4.6
+                return true;
```

Comment:
It's unit test and I assume most time it will run on machine with 4.6 runtime and above. I will change to assert.fail if it's more clear.